esti/catalog_export_test.go

```diff
@@ -383,3 +395,108 @@ func TestAWSCatalogExport(t *testing.T) {
+	resp, err := client.GetRepositoryWithResponse(ctx, testData.Repository)
+	require.NoError(t, err)
+	require.NotNil(t, resp.JSON200)
+	namespaceURL, err := url.Parse(resp.JSON200.StorageNamespace)
+	require.NoError(t, err)
+	keyTempl := "%s/_lakefs/exported/%s/%s/test_table/_delta_log/00000000000000000000.json"
+
+	switch blockstoreType {
+	case block.BlockstoreTypeS3:
+		cfg, err := config.LoadDefaultConfig(ctx,
+			config.WithRegion(testData.AWSRegion),
+			config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(testData.AWSAccessKeyID, testData.AWSSecretAccessKey, "")),
+		)
+		require.NoError(t, err)
+
+		clt := s3.NewFromConfig(cfg)
+		key := fmt.Sprintf(keyTempl, strings.TrimPrefix(namespaceURL.Path, "/"), mainBranch, commit[:6])
+		_, err = clt.HeadObject(ctx, &s3.HeadObjectInput{
+			Bucket: aws.String(namespaceURL.Host),
+			Key:    aws.String(key)})
+		require.NoError(t, err)
+
+	case block.BlockstoreTypeAzure:
+		azClient, err := azure.BuildAzureServiceClient(params.Azure{
+			StorageAccount:   testData.AzureStorageAccount,
+			StorageAccessKey: testData.AzureAccessKey,
+		})
+		require.NoError(t, err)
+
+		containerName, prefix, _ := strings.Cut(namespaceURL.Path, uri.PathSeparator)
+		key := fmt.Sprintf(keyTempl, strings.TrimPrefix(prefix, "/"), mainBranch, commit[:6])
+		_, err = azClient.NewContainerClient(containerName).NewBlobClient(key).GetProperties(ctx, nil)
+		require.NoError(t, err)
+
+	default:
+		t.Fatal("validation failed on unsupported block adapter")
+	}
+}
+
+func TestDeltaCatalogExport(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+	defer tearDownTest(repo)
+
+	accessKeyID := viper.GetString("access_key_id")
+	secretAccessKey := viper.GetString("secret_access_key")
+	testData := &exportHooksTestData{
+		Repository:            repo,
+		Branch:                mainBranch,
+		LakeFSAccessKeyID:     accessKeyID,
+		LakeFSSecretAccessKey: secretAccessKey,
+	}
+	blockstore := setupCatalogExportTestByStorageType(t, testData)
+
+	tmplDir, err := fs.Sub(exportHooksFiles, "export_hooks_files/delta")
+	require.NoError(t, err)
+	err = fs.WalkDir(tmplDir, "data", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if !d.IsDir() {
+			buf, err := fs.ReadFile(tmplDir, path)
+			if err != nil {
+				return err
+			}
+			uploadResp, err := uploadContent(ctx, repo, mainBranch, strings.TrimPrefix(path, "data/"), string(buf))
+			if err != nil {
+				return err
+			}
+			require.Equal(t, http.StatusCreated, uploadResp.StatusCode())
+		}
+		return nil
+	})
+	require.NoError(t, err)
+
+	headCommit := uploadAndCommitObjects(t, ctx, repo, mainBranch, map[string]string{
+		"_lakefs_actions/delta_export.yaml": renderTplFileAsStr(t, testData, tmplDir, fmt.Sprintf("%s/_lakefs_actions/delta_export.yaml", blockstore)),
+	})
+
+	runs := waitForListRepositoryRunsLen(ctx, t, repo, headCommit.Id, 1)
+	require.Equal(t, "completed", runs.Results[0].Status)
```

Comment:
Consider checking the run result name or other identifier to make sure you're seeing the correct result

Reply:
Done